cmd/mayactl/app/command/volume_info.go

```diff
@@ -166,63 +172,64 @@ Status  :   {{.Status}}
 		fmt.Println("Error displaying volume details, found error :", err)
 		return nil
 	}
-	replicaCount, _ = strconv.Atoi(a.ReplicaCount)
-	// This case will occur only if user has manually specified zero replica.
-	if replicaCount == 0 || a.ReplicaStatus == "" || a.Replicas == "" {
-		fmt.Println("None of the replicas are running, please check the volume pod's status by running [kubectl describe pod -l=openebs/replica --all-namespaces] or try again later.")
-		return nil
-	}
-	// Splitting strings with delimiter ','
-	replicaStatusStrings := strings.Split(a.ReplicaStatus, ",")
-	addressIPStrings := strings.Split(a.Replicas, ",")
-
-	// making a map of replica ip and their respective status,index and mode
-	replicaIPStatus := make(map[string]*Value)
-
-	for i, v := range addressIPStrings {
-		if v != "nil" {
-			replicaIPStatus[v] = &Value{index: i, status: replicaStatusStrings[i], mode: "NA"}
-		} else {
-			// appending address with index to avoid same key conflict
-			replicaIPStatus[v+string(i)] = &Value{index: i, status: replicaStatusStrings[i], mode: "NA"}
+	if strings.Contains(v.Spec.Iqn, "com.openebs.jiva") {
+		replicaCount, _ = strconv.Atoi(v.Spec.Replicas)
+		// This case will occur only if user has manually specified zero replica.
+		if replicaCount == 0 || v.ObjectMeta.Annotations["openebs.io/replica-status"] == "" {
```

Comment:
At this point, any property should be retrievable from casvolume itself. If you go by my earlier comments, casvolume that comes in from maya-apiserver can be updated at GetVolumeAnnotations. The original casvolume will be updated in such a way that later on it will be a unified retrieval.

Updating the original cas volume object implies, taking care of backware, forwards & different storage engine types.